core/src/main/scala/chisel3/experimental/package.scala

```diff
@@ -93,20 +96,27 @@ package object experimental {
         case _         => d
       }
     }
-    val ports: Seq[Data] =
-      gen._elements.toSeq.reverse.map {
-        case (name, data) =>
-          val p = chisel3.IO(coerceDirection(chiselTypeClone(data).asInstanceOf[Data]))
-          p.suggestName(name)
-          p
-
-      }
 
-    implicit val dv: DataView[Seq[Data], T] = DataView.mapping(
-      _ => chiselTypeClone(gen).asInstanceOf[T],
-      (seq, rec) => seq.zip(rec._elements.toSeq.reverse).map { case (port, (_, field)) => port -> field }
-    )
-    ports.viewAs[T]
+    type R = T with Record
+    gen match {
+      case _:      Element => IO(gen)
+      case _:      Vec[_] => IO(gen)
```

Comment:
Are we sure this is the correct behavior for Vec? I know it's unlikely people usually want ports named `0`, `1`, ... but this behavior is inconsistent with the behavior you'd get from `MixedVec` (which is actually a `Record).

Reply:
Discussed offline. Summarizing here for others.

It's unclear what should happen for `Vec[_]` other than some behavior should happen. Both `MixedVec` style behavior and the PR'd behavior (there is an `io` port) are reasonable. The PR'd behavior may be less surprising. If a user wants something different then they can create a bundle (or `MixedVec`) for that.